provider/source/src/time_zones/convert.rs

```diff
@@ -96,9 +98,91 @@ fn compute_meta_zone_ids_btreemap(
     for (meta_zone_id, meta_zone_id_data) in meta_zone_ids_resource.iter() {
         meta_zone_ids.insert(meta_zone_id_data.long_id.to_string(), *meta_zone_id);
     }
+    // TODO(#1781): Remove this special case once the short id is updated in CLDR
+    meta_zone_ids.insert("Yukon".to_owned(), MetazoneId(tinystr!(4, "yuko")));
     meta_zone_ids
 }
 
+fn offsets_for_meta(
+    meta_zones_territory: &Vec<MetazoneTerritory>,
+    meta: &str,
+    tzdb: &parse_zoneinfo::table::Table,
+) -> (UtcOffset, Option<UtcOffset>) {
+    let (mut standard, mut daylight) = (None, None);
+    let mut first_iana = "";
+    for m in meta_zones_territory {
+        if m.map_zone.other != meta {
+            continue;
+        }
+        let iana = &m.map_zone.zone_type;
```

Comment:
Issues/Suggestions:

1. `map_zone.zone_type` gives you only a single "golden zone" for the given metazone and territory. There might be more IANA zones in the same territory that you are missing. For example, I don't know if you ever reach a zone like America/Phoenix with this code.
2. IANA zones can switch between metazones, and you don't account for that here. You can see in the data when a zone switches metazones. You should ideally read the TZDB entries only for the period in which the zone was in the metazone.
3. I think a better approach might be to flip this around: start with the IANA zone, loop over all metazones, load the offsets for the given time period, and then insert them into a map from metazones to offsets. By the end, you will have comprehensively touched all time zones and metazones.